packages/dds/map/src/directory.ts

```diff
@@ -2215,10 +2215,15 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 					localOpMetadata !== undefined && isKeyEditLocalOpMetadata(localOpMetadata),
 					0x011 /* pendingMessageId is missing from the local client's operation */,
 				);
-				assert(
-					pendingKeyMessageIds[0] === localOpMetadata.pendingMessageId,
-					0x331 /* Unexpected pending message received */,
-				);
+				if (pendingKeyMessageIds[0] !== localOpMetadata.pendingMessageId) {
+					// TODO: AB#7742: Hitting this block indicates that the pending message Id received
```

Comment:
I don't know if this is enough information to track the issue once it happens. One thing I think we can add is this.pendingClearMessageIds.length to see if it's related to that case. I hadn't look deeply into it but I guess there is more info we can add.

Reply:
Yes, we do have similar logic for validating whether the `pendingClearMessageId` matches the one in `localOpMetadata`. However this is related to 0x32a, and we haven't observed many hits yet. I guess we can log that error as well once we receive complaints or encounter numerous assertion hits.

Reply:
i think logging length is a good idea as well.